ops/fhir_validations/validations.py

```diff
@@ -0,0 +1,178 @@
+        file_filters = get_file_filter(ignore_list, file_name)
+
+        for error in errors_per_file[file_name]:
+            if not any_filters_match(global_filters, error) and not any_filters_match(file_filters, error):
+                filtered_errors_per_file.setdefault(file_name, []).append(error)
+
+    return filtered_errors_per_file
+
+
+def validate_resources(validator_path: str, version: str, ignore_list: dict, files: List[str]) -> dict:
+    java_commands = ['java', '-Xmx3G', '-Xms2G', '-jar', validator_path]
+
+    for file_name in files:
+        java_commands.append(file_name)
+
+    java_commands.append('-version')
+    java_commands.append(version)
+
+    java_call = subprocess.run(java_commands, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    output = java_call.stdout.decode('utf-8')
+    error_output = java_call.stderr.decode('utf-8')
+    if java_call.returncode != 0:
+        if error_output != '':
+            print(error_output)
+            print('There was an issue processing the request.')
+            exit(1)
+        elif output == '':
+            print('Validation failed, but no output was generated.')
+            exit(1)
+    output_lines = output.split('\n')
+
+    file_name = "loading_output"
+    errors_per_file = {}
+
+    for line in output_lines:
+        if line.startswith('-- '):
+            file_search = re.search('-- ([^\\s]*) -*', line)
+
+            if file_search:
+                file_name = file_search.group(1)
+                errors_per_file[file_name] = []
+
+        stripped_line = line.strip()
+
+        if stripped_line.lower().startswith('error @'):
+            errors_per_file[file_name].append(stripped_line)
+
+    return filter_errors(ignore_list, errors_per_file)
+
+
+class RunConfig(object):
+    def __init__(self):
+        self.target_dir = ''
+        self.version = ''
+
+
+def validate_resource_dir(validator_path: str, run_config: RunConfig, ignore_list: dict, recently_changed: bool) -> Tuple[int, dict]:
+    print('Checking directory {}'.format(run_config.target_dir))
+    files = get_fhir_resource_files(run_config.target_dir, recently_changed)
+    files.sort()
+    file_count = len(files)
+
+    if file_count > 0:
+        print('Validating {} resources'.format(file_count))
+
+        invalid_resources = validate_resources(validator_path, run_config.version, ignore_list, files)
+        for resource in invalid_resources:
+            if len(resource) == 0:
+                del invalid_resources[resource]
+    else:
+        print('No resources to validate')
+        invalid_resources = {}
+
+    return file_count, invalid_resources
+
+
+def main():
+    args = diff_check.parse_arguments()
+
+    try:
+        ignore_list = yaml.safe_load(open(args.ignorefile))
+        print('Ignore list found, using to filter results')
+    except FileNotFoundError:
+        print('Could not find ignore list file, running without filters')
+        ignore_list = {'ignore_list': {}}
```

Comment:
🍰 Looking at the ignore list, this case will assuredly mean a failure, right? we have some unfixable errors in there, which means we can never pass if we fail to load the ignore list. If so, perhaps we should just return in this error case and not try to continue?

Reply:
If you wanted to just run without any ignores to see how it's currently looking, you could just specify no ignore list, so there is a valid run case for this path.